src/meta-srv/src/procedure/region_follower/remove_region_follower.rs

```diff
@@ -67,11 +67,35 @@ impl RemoveRegionFollowerProcedure {
         // loads the datanode peer and check peer is alive
         self.data.peer = self.data.load_datanode_peer(&self.context).await?;
 
-        // loads the datanode table value
-        self.data.datanode_table_value = self.data.load_datanode_table_value(&self.context).await?;
+        // loads the table route of the region
+        self.data.table_route = self.data.load_table_route(&self.context).await?;
 
         // loads the table route of the region
         self.data.table_route = self.data.load_table_route(&self.context).await?;
+        let region_leader_datanode_id = {
+            // Safety: load table route is done before.
+            let table_route = self.data.physical_table_route().unwrap();
+            table_route
+                .region_routes
+                .iter()
+                .find(|region_route| region_route.region.id == self.data.region_id)
+                .context(error::RegionRouteNotFoundSnafu {
+                    region_id: self.data.region_id,
+                })?
+                .leader_peer
+                .as_ref()
+                .context(error::UnexpectedSnafu {
```

Comment:
```suggestion
                .with_context(|| error::UnexpectedSnafu {
```

Reply:
resolved in [bb483d8](https://github.com/GreptimeTeam/greptimedb/pull/5781/commits/bb483d8e9c6404a653cb00732573546df3a028a0)